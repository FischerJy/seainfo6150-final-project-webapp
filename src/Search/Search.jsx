import React, { createRef, useEffect, useState } from "react";
import { useHistory } from "react-router-dom";

export default function Search() {
  const node = createRef();
  const his = useHistory();

  return (
    <div>
      <br />
      <br />
      <p style={{ textAlign: "center" }}>
        name: <input type="text" ref={node} />
        <button
          onClick={() => {
            his.push("/search/" + node.current.value);
          }}
        >
          Search
        </button>
      </p>
    </div>
  );
}
