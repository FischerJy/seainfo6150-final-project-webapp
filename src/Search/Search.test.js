import React from "react";
import { render } from "@testing-library/react";
import Search from "./Search.jsx";
import { BrowserRouter } from "react-router-dom";



it("renders the Search component correctly", () => {
    const { container } = render(<BrowserRouter>
        <Search />
    </BrowserRouter>);
    expect(container).toMatchSnapshot();
});