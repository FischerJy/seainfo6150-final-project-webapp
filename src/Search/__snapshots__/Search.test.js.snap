// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Search component correctly 1`] = `
<div>
  <div>
    <br />
    <br />
    <p
      style="text-align: center;"
    >
      name: 
      <input
        type="text"
      />
      <button>
        Search
      </button>
    </p>
  </div>
</div>
`;
